pkg/sql/compile/ddl.go

```diff
@@ -90,6 +89,29 @@ func (s *Scope) CreateTable(c *Compile) error {
 	return colexec.CreateAutoIncrCol(dbSource, c.ctx, c.proc, planCols, tblName)
 }
 
+func (s *Scope) TruncateTable(c *Compile) (uint64, error) {
+	tqry := s.Plan.GetDdl().GetTruncateTable()
+	dbName := tqry.GetDatabase()
+	dbSource, err := c.e.Database(c.ctx, dbName, c.proc.TxnOperator)
+	if err != nil {
+		return 0, err
+	}
+	tblName := tqry.GetTable()
+	var rel engine.Relation
+	if rel, err = dbSource.Relation(c.ctx, tblName); err != nil {
+		return 0, err
+	}
+	affect, err := rel.Truncate(c.ctx)
```

Comment:
We said truncate in 0.6 need to generate a new table id.   Where does this happen?

Reply:
The front end only calls a truncate interface provided by Mochen. This interface does not belong to the tae, and its internal implementation is transparent to me. The tableId needs to be constructed as the implementation of the truncate parameter, which depends on some objects inside the engine. I can't get it here